### PR TITLE
python3Packages.xdg: 4.0.1 -> 5.0.1

### PR DESCRIPTION
**Pull request by jonringer**
###### Motivation for this change
routine bump of packages I maintain

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jonringer** [comment]
```
https://github.com/NixOS/nixpkgs/pull/109074

4 packages built:
python37Packages.goobook python37Packages.xdg python38Packages.goobook python38Packages.xdg
```

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 109074` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python37Packages.goobook</li>
    <li>python37Packages.xdg</li>
    <li>python38Packages.goobook</li>
    <li>python38Packages.xdg</li>
  </ul>
</details>

File: pkgs/development/python-modules/xdg/default.nix [hunk 1]
```diff
@@ -14,7 +14,7 @@ buildPythonPackage rec {
     owner = "srstevenson";
     repo = pname;
     rev = version;
-    sha256 = "13kgnbwam6wmdbig0m98vmyjcqrp0j62nmfknb6prr33ns2nxbs2";
+    sha256 = "sha256-WMY9Hs9SbErTp8hVBoCWTz4dLQhuzqTRpXFEyE7+woo=";
   };
 
   nativeBuildInputs = [ poetry ];
```
Comment (fabaff) [inline pkgs/development/python-modules/xdg/default.nix:20]: Upstream switched to [`poetry-core`](https://github.com/srstevenson/xdg/releases/tag/5.0.1).

```suggestion
  nativeBuildInputs = [ poetry-core ];
```

Also, requires an update of [line 3](https://github.com/NixOS/nixpkgs/pull/109074/files#diff-d8f40ff218187282ffe449456775e5eb6145275e2b1dfa99ef5b377dce112334L3).

Comment (jonringer) [reply pkgs/development/python-modules/xdg/default.nix:20]: thanks for the catch

**Pull request merged**